### DISCUSSION TITLE
Type file for Yazi's API

### DISCUSSION THREAD

**Discussion by hankertrix** [sxyazi/yazi#2121] [2. Show and tell]
Hopefully, this will be useful to other plugin developers. I made it to have type-checking for my plugin, which is probably the most complex plugin currently available for Yazi, so static typing is a must to catch bugs.

You can find it here:
https://github.com/hankertrix/augment-command.yazi/blob/main/yazi_types.lua

To use it, you need to have [Lua Language Server](https://github.com/LuaLS/lua-language-server) installed and set up and have the file in the directory where you are developing your plugin. It doesn't need to be included in your plugin.

I only typed the parts of the API that are documented, or that I know how to use. Yazi exposes quite a bit more, but I have no idea how to use those APIs, so it's impossible to type them.

I will try to update it to keep up with Yazi API changes, but I won't have much time to do so after this month, so it may be slow to update and might be outdated at some points.

**Comment by AnirudhG07** [comment]
Cool! Can you add it somewhere in [awesome-yazi](https://github.com/AnirudhG07/awesome-yazi) maybe under `developer utilities` ( you can add it). This way more people will know about it and use it!

**Reply by hankertrix** [reply]
Oh sure, I'll submit a PR.

**Reply by hankertrix** [reply]
PR submitted: https://github.com/AnirudhG07/awesome-yazi/pull/38

**Comment by ram-xv** [comment]
@hankertrix thanks for this!!!

**Reply by hankertrix** [reply]
You're welcome!
